doc/manual/src/glossary.md

```diff
@@ -208,6 +208,44 @@
 
     See [installables](./command-ref/new-cli/nix.md#installables) for [`nix` commands](./command-ref/new-cli/nix.md) (experimental) for details.
 
+  - [build system]{#gloss-build-system}
+
+    [build system]: #gloss-build-system
+
+    A representation of the processor architecture and platform that performs the build or compilation, typically a dash-separated string.
+
+    This is based on the [GCC system terminology](#gloss-gcc-system-terminology).
+
+    Programs that are produced by a [derivation] were configured to have a [build system] that can be executed by the [derivation system](language/derivations.md#attr-system). When using the program, only its [host system] is still relevant.
+
+  - [host system]{#gloss-host-system}
+
+    [host system]: #gloss-host-system
+
+    A representation of the processor architecture and platform where build products will run on, typically a dash-separated string.
+
+    This is based on the [GCC system terminology](#gloss-gcc-system-terminology).
+
+    Programs that run natively in a [derivation] must be configured with a [host system] that can by executed by the [derivation system](language/derivations.md#attr-system).
+
+  - [target system]{#gloss-target-system}
+
+    [target system]: #gloss-target-system
+
+    A representation of the processor architecture and platform that a compiler will produce, typically a dash-separated string.
+
+    When discussing a build that does not produce a compiler, we only use [build system] and [host system] to describe it; we try to avoid discussing the actual build from the perspective its compiler's configuration.
```

Comment:
```suggestion
    When discussing a build that does not produce a compiler, we only use [build system] and [host system] to describe it.

    We try to avoid discussing the actual build from the perspective its compiler's configuration, and instead always talk about the build from the perspective of the build itself.
    Avoiding that ensures that if we are building a compiler do use "[target system]", it is clear we are referring to the target of the compiler being built, and not the target of a preexisting compiler used during the build.
```
I don't know if that is too much a digression for Nix, but I do like to make the ambiguity we're avoiding explicit, rather than left to the reader's imagination :).

Reply:
```suggestion
    When discussing a build that does not produce a compiler, we only use [build system] and [host system] to describe it.
    
    We try to avoid discussing the actual build from the perspective of its compiler, instead focusing on the build itself.
    Otherwise you won't have a term to distinguish the actual [target system] from what you should have called the [host system].
```

I think I've managed to simplify it. Not sure if I should have tried this in the morning :sweat_smile:

Reply:
@Ericson2314 wdyt?